src/retractions.jl

```diff
@@ -706,8 +706,13 @@ function retract_pade! end
     retract_project!(M::AbstractManifold, q, p, X)
 
 computes the mutating variant of the [`ProjectionRetraction`](@ref).
+By default this [`project`](@ref)s `X` onto the manifold (in place of `q`)
+after [`embed!`](@ref)ing `X`.
 """
-retract_project!(M::AbstractManifold, q, p, X)
+retract_project!(M::AbstractManifold, q, p, X) = project!(M, q, embed(M, p, X))
```

Comment:
```suggestion
retract_project!(M::AbstractManifold, q, p, X) = project!(M, q, p + embed(M, p, X))
```
Shouldn't it be more like that?

Reply:
Do we have a + in the embedding? Not in general.

Reply:
Well, `retract_project!(M::AbstractManifold, q, p, X) = project!(M, q, embed(M, p, X))` seems incorrect -- when `X` is a zero vector, retraction should just return `p`, and your version won't do that. So I think we can either accept that this only works when we have `+` in the embedding or not have this default definition at all?

Reply:
Then, maybe, no default at all is what I would prefer, I think